### ISSUE TITLE
MueLu:  Can default GO macros depend on default types in Tpetra?

### ISSUE THREAD

**Issue by jhux2** [trilinos/Trilinos#5932]
@trilinos/muelu @csiefer2 

As reported by @tjfulle:

"The situation:

```
cmake -DTrilinos_ENABLE_Tpetra=ON  -DTpetra_ENABLE_DEPRECATED_CODE=OFF [PR options to enable all downstream, tests etc.] -DMueLu_ENABLE_Epetra=OFF ${TRILINOS_DIR}
```

I would expect (hope?) that MueLu would inherit Tpetra’s default global ordinal, but with this config `muelu/src/Headers/MueLu_Details_DefaultTypes.hpp` set MueLu’s default global ordinal to int.  I *think* this is causing link errors in many examples/tests (though I can’t be sure because the linker spews hundreds of pages of template vomit).

Can the macros `HAVE_MUELU_DEFAULT_GO_*` be related to the default types in Tpetra, rather than preferring int?"

**Comment by tjfulle** [comment]
Thanks @jhux2!  I forgot to add that this is with `-DTpetra_ENABLE_DEPRECATED_CODE=OFF`

**Comment by jhux2** [comment]
@tjfulle Can you tell us what is reported during cmake configure in the MueLu section that starts with `MueLu ETI support enabled`?

**Comment by tjfulle** [comment]
@jhux2, ya.  I'm in the middle of a build right now, give me a bit until it fails again :)

**Comment by brian-kelley** [comment]
@tjfulle @jhux2 Yes, this is an issue with my changes from #5697. I'm working on a fix.

**Comment by jhux2** [comment]
@trilinos/MUELU

**Comment by brian-kelley** [comment]
The fix in #5933 seems to have worked, so I'm closing.
